File: aoc2022/day001/solution.py
def solve1(inputfile):
    inn = open(inputfile, "r").read()
    I = [x.split("\n") for x in inn.split("\n\n")]
    I = [sum([int(y) for y in x]) for x in I]

    return max(I)


def solve2(inputfile):
    inn = open(inputfile, "r").read()
    I = [x.split("\n") for x in inn.split("\n\n")]
    I = [sum([int(y) for y in x]) for x in I]

    return sum(sorted(I)[-3:])


if __name__ == "__main__":
    TEST = "test.txt"
    INPUT = "input.txt"

    print("-- TEST --")
    print(solve1(TEST))
    print(solve2(TEST))

    print("-- SOLUTION --")
    print(solve1(INPUT))
    print(solve2(INPUT))

File: aoc2022/day001/solution.scala
@main def main =
    import scala.io.Source
    import scala.math.Ordering

    def parseInput(inputfile: String): Array[Array[Int]] =
        val inn = Source.fromFile(inputfile).mkString
        inn.split("\n\n").map(_.split("\n").map(_.toInt))

    def solve1(inputfile: String): Int =
        val I = parseInput(inputfile)
        I.map(_.sum).max

    def solve2(inputfile: String): Int =
        val I = parseInput(inputfile)
        I.map(_.sum)
            .sorted(Ordering.Int.reverse)
            .take(3)
            .sum

    val test = "test.txt"
    val input = "input.txt"

    println("-- TEST --")
    println(solve1(test))
    println(solve2(test))

    println("-- SOLUTION --")
    println(solve1(input))
    println(solve2(input))